### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by arnavs**
Replaced the two problems with a unified example which makes the problems in comments obsolete.

### CODE REVIEW

**Comment by frapac** [comment]
Thanks for this update!

File: README.md [hunk 1]
```diff
@@ -30,39 +30,34 @@ MathOptInterface Interface
...
-@NLconstraint(m, sqrt(x[1]^2+x[2]^2) <= 1)
-@show JuMP.optimize!(m)
+optimize!(m)
+(value(x), value(y), value(z), value(u), objective_value(m), termination_status(m)) # (5)
 ```
 
-The `MathProgBase` interface has been deprecated. 
+1. Setting `KNITRO` options. 
+2. Setting initial conditions on variables. 
+3. Setting box bounds on variables.
```
Comment (frapac) [inline README.md]: I would rephrase a little here: `Setting box constraints on variables`

File: README.md [hunk 2]
```diff
@@ -30,39 +30,34 @@ MathOptInterface Interface
...
+@variable(m, x, start = 1.2) # (2)
+@variable(m, y)
+@variable(m, z)
+@variable(m, 4.0 <= u <= 4.0) # (3)
+
+mysquare(x) = x^2 
+register(m, :mysquare, 1, mysquare, autodiff = true) # (4)
 
-m = Model(with_optimizer(KNITRO.Optimizer))
+@NLobjective(m, Min, mysquare(1 - x) + 100(y-x^2)^2 + u)
```
Comment (frapac) [inline README.md]: Is it possible to add spaces before and after the minus term? 
`(y - x^2)^2`
In general, I try to follow JuMP's conventions:
http://www.juliaopt.org/JuMP.jl/v0.20.0/style/#Whitespace-1

File: README.md (hunk 1 above)
Comment (arnavs) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (arnavs) [reply README.md]: Of course. Done.

File: README.md [hunk 3]
```diff
@@ -30,39 +30,34 @@ MathOptInterface Interface
...
+@variable(m, x, start = 1.2) # (2)
+@variable(m, y)
+@variable(m, z)
+@variable(m, 4.0 <= u <= 4.0) # (3)
+
+mysquare(x) = x^2 
+register(m, :mysquare, 1, mysquare, autodiff = true) # (4)
 
-m = Model(with_optimizer(KNITRO.Optimizer))
+@NLobjective(m, Min, mysquare(1 - x) + 100(y - x^2)^2 + u)
```
Comment (frapac) [inline README.md]: Almost good! I would just add a `*` before the parenthesis (explicit is better than implicit): 
`100 * (y - x^2)^2`

Comment (arnavs) [reply README.md]: OK

Comment (arnavs) [reply README.md]: Done
